src/Transports/MassTransit.AmazonSqsTransport/Pipeline/AmazonSqsMessageReceiver.cs

```diff
@@ -68,13 +69,12 @@ async Task Consume()
             catch (Exception exception)
             {
                 LogContext.Error?.Log(exception, "Consume Loop faulted");
+                throw;

```

Comment:
Why do you need throw? you will log, and complete the loop, all transports work this way.
https://github.com/MassTransit/MassTransit/blob/71dad3b5ec9c658674f2e674f0052f617fb22b7c/src/Transports/MassTransit.AmazonSqsTransport/Pipeline/AmazonSqsMessageReceiver.cs#L117 it is infinite loop (until stopped)

Reply:
And what happens to the exception? If the exception is not propagated how would the ReceiveTransport know it needs to retry?

Reply:
I think the best way to do it:
https://github.com/MassTransit/MassTransit/blob/b44eb5e6acdfdfce37192a42a40f87b4b2dd4622/src/Transports/MassTransit.KafkaIntegration/Transport/KafkaMessageReceiver.cs#L112

otherwise, `Exception` is could not been retried

Reply:
I'd add additional catch in here (specific for SQS exception) and mark completed task as faulted:
https://github.com/MassTransit/MassTransit/blob/9e21c29bba03ddb96d6eb17117c79e8691a85a43/src/Transports/MassTransit.KafkaIntegration/Exceptions/KafkaConnectionException.cs#L31

Reply:
This is a different way of doing the same, but not necessary better as the consume look is not technically finished. If I did it the way you suggested, I would be completing the agent while a thread running the Consume task loop is technically still running. There's a chance future changes will add things to the finally block that are either blocking or end up in undesired concurrency. By tying the completion of the agent to the completion of the consume loop, I ensure all the work completes cleanly before giving ReceiveTransport a chance to handle to the completion, be it caused by error or not.

Reply:
There is tricky part is `Task.WhenAll()` but it could be handled by other `CancellationToken` which should be cancelled by any exception. With your approach result will be the same apart from "We do not re throw in exception block in general for Receiver", you have to mark Completed as fail instead.
I remember throw leads kafka to unpredictable results in Health checks, that's why I encourage to set Completed task instead.
There are 2 situations you have to keep in mind:
1. Transient connectivity issues, and transport could be recovered
2. Transport recycle

@phatboyg your thoughts?

Reply:
Regard adding an additional catch, it is only applicable if I'd change the implementation to call SetCompleted in the catch block -- which I am against as per my above comment. But even so, it would just duplicate the error handling logic which would already exist in the Consume function. Generally, it is a good practice to catch exceptions at the highest level and only add error handling in places where you can change the outcome.

Reply:
@NooNameR I agree the handling of cancellation is not ideal in the ReceiveMessage function. I am not touching it because my goal here is to fix a critical bug which currently makes this transport unreliable for production use. I think improvements unrelated to the issue are better discussed separately. 

IMHO this PR needs to be merged and released ASAP, I had 2 prod issues this week because of this issue, one in which 10 services collectively stopped polling SQS after a network failover.